File: frontend/app/api/transactions/deviceModels/updateDeviceModelTransaction.ts
import { API_BASE_URL } from "../../client/apiClient"
import { UpdateDeviceModelType } from "../../../types/deviceModelTypes"
import { getDeviceModelsFromApi } from "../../deviceModels/fetchDeviceModels"
import {
         normalizeDeviceModel,
         toUpdateDeviceModelRequest
       } from "../../../utils/deviceModelMapper"
import { authFetch } from "../../client/apiClient"

type UpdateDeviceModelTransactionParams = {
                                             deviceModel: UpdateDeviceModelType
                                             setDeviceModels: any
                                           }

export async function updateDeviceModelTransaction({
                                                     deviceModel,
                                                     setDeviceModels
                                                   }: UpdateDeviceModelTransactionParams
                                                 )
{
  console.log("updateDeviceModelTransaction")


  await authFetch(
                `${API_BASE_URL}/update-device-model`,
                {
                  method: "POST",
                  headers: {
                            "Content-Type":
                            "application/json"
                            },
                  body: JSON.stringify(
                                          toUpdateDeviceModelRequest(
                                                                        deviceModel
                                                                      )
                                        )
                }
              )

  const deviceModels =
    await getDeviceModelsFromApi()

  setDeviceModels(
                    deviceModels.map(
                                      normalizeDeviceModel
                                    )
                  )
}

File: backend/transactions/device_models/update_device_model_transaction.py
from device_models.update_device_model import update_device_model
from schemas.device_model_schemas import UpdateDeviceModelRequest
from supabase import Client

def update_device_model_transaction(
                                        client:Client,
                                        device_model: UpdateDeviceModelRequest,
                                        hospital_id: str
                                    ):

    print("update_device_model_transaction")

    return update_device_model(
                                client,
                                device_model,
                                hospital_id
                              )